### PR TITLE
Web3 status updates

### PR DESCRIPTION
**Pull request by r-czajkowski**
~Depends on: #2101~
Closes: #2057
Closes: #2026 

This PR updates the web3 status indicator in the KEEP token dashboard.

obraz

obraz

obraz

obraz

obraz

obraz

### CODE REVIEW

**Comment by Shadowfiend** [comment]
🤔 is this based on any existing designs, or is it groundwork for implementing designs? This is what I'm seeing in Figma:

Screen Shot 2020-10-06 at 11 39 11

But I'm also seeing that the referenced issue only mentions capitalization, not any updates to styling or positioning of information.

**Comment by liz-shinn** [comment]
Yes, the design for this wallet section had updated since these screenshots were taken, based on feedback from @r-czajkowski and @Gluzman. The styling you've screenshotted from Figma is the correct one.

**Comment by liz-shinn** [comment]
Went into more detail on some different wallet states in this issue: https://github.com/keep-network/keep-core/issues/2026

**Comment by liz-shinn** [comment]
>Not sure what we want to display when a user is not connected with a wallet.

@r-czajkowski Do you mean what we display in the body of the dapp?

The Overview page would be what is shown instead of the wallet wall page, even when there's no wallet connected. Instead of numerical data we can show some empty states. https://github.com/keep-network/keep-core/issues/2002

**Comment by liz-shinn** [comment]
Noting that title of this page shouldn't be "Connect wallet" — should instead be "Overview" and show this:

Onboarding

**Comment by liz-shinn** [comment]
I updated the Delegation page (for both Granted and Token subnavigation) to show an empty state for when a wallet isn't connected. 

Delegation Empty State


I don't think this is a page where it makes sense to show the full content with a bunch of N/A everywhere. I think it's better to explain what will be there (with a graphic) and give a clear action (Connect a Wallet).

@Shadowfiend @r-czajkowski

**Comment by liz-shinn** [comment]
Sidenote that the case of the network status should be consistent. If we're using title case for "Network Disconnected," we should use title case for the other network statuses as well.

"Private"
"Please Connect a Wallet"
etc.

**Comment by liz-shinn** [comment]
An empty state page for Token Grant page when a wallet isn't connected:

Grant Empty State

**Comment by r-czajkowski** [comment]
> Updated the Delegation page to show an empty state for when a wallet isn't connected. Was thinking these could be the same for both Granted and Wallet subnavigation. Artboard in Figma file: https://www.figma.com/file/Su61MwHLrQn3ld24q85pMx/Token-Dashboard-dApp?node-id=6710%3A10525
> 
> Delegation Empty State
> 
> I don't think this is a page where it makes sense to show the full content with a bunch of N/A (Not Available) for all the numbers and data. It also doesn't make sense to show 0s when a wallet isn't connected since that's a separate state (a zero balance =/= disconnected wallet).
> 
> Would recommend explaining what _will_ be there with a graphic and give a clear action (Connect a Wallet).
> 
> @Shadowfiend @r-czajkowski

Totally agree, this empty states are awesome! 🚀 I'll open an issue for that and add to the milestone!

File: solidity/dashboard/src/components/AddressShortcut.jsx [hunk 1]
```diff
@@ -1,25 +1,26 @@
...
-      className={`address-shortcut tooltip address ${classNames}`}
-    >
-      <span className="tooltip-text address bottom">{copyStatus}</span>
-      {shortenAddress(address)}
-    </span>
+    <CopyToClipboard
+      toCopy={address}
+      render={({ reset, copyStatus, copyToClipboard }) => {
+        return (
+          <span
```
Comment (Shadowfiend) [inline solidity/dashboard/src/components/AddressShortcut.jsx:13]: `button` if it's clickable, as a general rule.

File: solidity/dashboard/src/components/NetworkStatus.jsx [hunk 2]
```diff
@@ -1,24 +1,27 @@
...
+import * as Icons from "./Icons"
 
 export const NetworkStatus = () => {
-  const { networkType, provider, error, yourAddress } = useContext(Web3Context)
+  const { networkType, provider, error, yourAddress } = useWeb3Context()
+
+  let status = "disconnected"
+  if (error) {
+    status = "error"
+  } else if (yourAddress && !error && provider !== null) {
```
Comment (Shadowfiend) [inline solidity/dashboard/src/components/NetworkStatus.jsx:11]: This feels like something the web3 context should expose directly as an `isConnected` or something, though I'm not sure 🤔

File: solidity/dashboard/src/components/NetworkStatus.jsx [hunk 3]
```diff
@@ -1,24 +1,27 @@
...
       />
-      <h5 className="text-grey-50">
-        {!yourAddress && "not connected"}
-        {yourAddress &&
-          !error &&
-          provider !== null &&
-          `connected: ${networkType}`}
-        {yourAddress && error && provider !== null && `wrong network`}
-      </h5>
+      <div className={`network-status__text--${status}`}>
```
Comment (Shadowfiend) [inline solidity/dashboard/src/components/NetworkStatus.jsx:20]: Is this truly a `div`? Could we get away with no element or a `span`? It doesn't feel block-level at first blush.

**Review by Shadowfiend — APPROVED** [review]
I'm approving here, but the merge conflict isn't easily resolvable so I couldn't do a test run. I think once you handle the merge conflict you can merge the PR, as this functionality is central enough that I'll test it in-browser in any of the next PRs anyway.

**Pull request merged**